botorch/acquisition/input_constructors.py

```diff
@@ -130,19 +135,32 @@ def construct_inputs_analytic_base(
         model: The model to be used in the acquisition function.
         training_data: A TrainingData object contraining the model's
             training data. `best_f` is extracted from here.
-        objective: The objective to in the acquisition function.
+        posterior_transform: The posterior transform to be used in the
+            acquisition function.
 
     Returns:
         A dict mapping kwarg names of the constructor to values.
     """
-    return {"model": model, "objective": objective}
+    # TODO: remove **kwargs once the AcquisitionObjective is removed.
+    if "objective" in kwargs:
+        warnings.warn(
+            "The `objective` argument to AnalyticAcquisitionFunctions is deprecated "
+            "and will be removed in the next version. Use `posterior_transform` "
+            "instead.",
+            DeprecationWarning,
+        )
+    return {
+        "model": model,
+        "posterior_transform": posterior_transform,
+        **{k: v for k, v in kwargs.items() if k == "objective"},
+    }
```

Comment:
Would you want to instantiate `posterior_transform` here from the objective, as you do in the deprecation helper? Doing this and raising the warning here would make more sense than deferring that to the constructor.